### PR TITLE
Fix: Tests hanging on MacOS when testing error on phase

### PR DESCRIPTION
**Pull request by jromero**
Signed-off-by: Javier Romero <jromero@pivotal.io>

### CODE REVIEW

**Comment by jromero** [comment]
> I think it's important that the tar writer be closed before the pipe. Otherwise, the tar footer will be cut off: https://golang.org/src/archive/tar/writer.go?s=14125:14156#L456

Interesting, was not aware that the footer would be written on `Close`. Will investigate further, including any potential alternative solutions.

**Comment by jromero** [comment]
@sclevine I dug into a bit further and it seems to be that the issue only arises when we call `tw.Close()` when an error while writing has occurred.

I've made additional changes to address your concerns as well. Let me know what you think and if the solution makes sense.

File: internal/archive/archive.go [hunk 1]
```diff
@@ -34,11 +34,16 @@ func readAsTar(src, basePath string, uid, gid int, mode int64, writeFn func(tw *
 	go func() {
 		var err error
 		defer func() {
-			w.CloseWithError(err)
+			_ = w.CloseWithError(err)
```
Comment (ekcasey) [inline internal/archive/archive.go]: apologies for the nitpick, but if we aren't using any of the return values I think we should omit the assignment to `_`

File: internal/archive/archive.go [hunk 2]
```diff
@@ -34,11 +34,16 @@ func readAsTar(src, basePath string, uid, gid int, mode int64, writeFn func(tw *
...
 		}()
 
 		tw := tar.NewWriter(w)
-		defer tw.Close()
+		defer func() {
+			// only close if no errors have occurred
+			if err == nil {
+				_ = tw.Close()
```
Comment (ekcasey) [inline internal/archive/archive.go]: same thing

**Pull request merged**